opentelemetry-sdk/src/opentelemetry/sdk/metrics/__init__.py

```diff
@@ -147,6 +153,17 @@ def __init__(
             monotonic=monotonic,
         )
 
+    def add(
+        self, label_values: Sequence[str], value: metrics_api.ValueT
+    ) -> None:
+        """See `opentelemetry.metrics.Counter.add`."""
+        self.get_handle(label_values).add(value)
+
+    def update(
+        self, label_values: Sequence[str], value: metrics_api.ValueT
+    ) -> None:
+        self.add(label_values, value)
+
 
 class Gauge(Metric):
```

Comment:
`Gauge` doesn't extend `metrics_api.Gauge`? There may be a better way to structure these classes, which might also save the redundant `update` below.

Reply:
The sdk separation is weird because each metric `counter`, `gauge` and `measure` want to use a generic `get_handle` implementation which is defined in the base class `Metric` in the SDK. But they also want to implement the API methods defined in `metric_api.Gauge` and such.

If we want to get rid of "update" and have inheritance that makes sense, we can simply implement a `get_handle` for each metric type. Yes it will have duplicates of `get_handle`, but with this we can get rid of `HANDLE_TYPE`.

Also, I'm not sure how to get rid of `update`, as it is used in `record_batch` to do generic updates to metrics regardless of metric type. One I can think of is to have a static field for each metric type pointing to the corresponding function for each metric type (`add`, `set`, `record`).
Thoughts?

Reply:
Wouldn't multiple inheritance still work in this case? If you inherit such that behavioral classes come after interface classes:

`class Gauge(Metric, metrics_api.Gauge)`

It should pick up the first parent in the MRO that has the method in question, and use that. 

That still looks a little weird but does send the message that the interface works as expected. +1 that if we get rid of the API/SDK separation we can do this slightly cleaner.

Regarding update: a static field or just re-implementing seems fine to me. sometimes verbosity is better, it often is for the reader.

Reply:
That makes sense. I guess this can be a solution for now until we decide how to handle the API/SDK shenanigans.